### PR TITLE
Checkbox dispatchers

### PR DESCRIPTION
**Pull request by sephcoster**
Filter checkboxes dispatch appropriate actions to call reducers to adjust query state to represent the newly selected values in the query.

## Additions

- FILTER_CHANGED action
- Query reducer updates as a result of filter change action
- AggregationItem component using redux / connect

## Removals

-

## Changes

- Query reducer code was changed with the addition of the filter change function

## Testing

- Coverage remains at 100%.

## Review

- @JeffreyMFarley 

[Preview this PR without the whitespace changes](?w=0)

## Notes

-

## Todos

- Update default state to first check URL structure so state remains the same on page refresh and links can be sent to others to duplicate the filter criteria.

## Checklist

* [ X ] Changes are limited to a single goal (no scope creep)
* [ X ] Code can be automatically merged (no conflicts)
* [ X ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ X ] Passes all existing automated tests
* [ X ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ X ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: config/webpack.config.dev.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ module.exports = merge({
         loader: 'string-replace',
         query: {
           search: '@@API',
-          replace: 'http://localhost:8000/data-research/consumer-complaints/api/v1/'
```
Comment (JeffreyMFarley) [inline config/webpack.config.dev.js]: um... how is this working for you?

I am running `cfgov-refresh` and that path is where the API is running.  What does your setup look like?

Comment (sephcoster) [inline config/webpack.config.dev.js]: I'm running the API as its own Django app separate from refresh as I was having trouble with flags / routing for a bit with the UI injection.

 `npm start` for the UI, `runserver` for the API to keep things flexible locally without touching CF-gov just yet.

**Comment by JeffreyMFarley** [comment]
Can you merge with master? I see that you have my commits, but they are not from the same tree...

https://github.com/cfpb/ccdb5-ui/network
screen shot 2017-06-30 at 10 48 59 am

**Comment by sephcoster** [comment]
Sure - can do.

**Pull request merged**